rllib/utils/replay_buffers/replay_buffer.py

```diff
@@ -63,9 +64,102 @@ def warn_replay_capacity(*, item: SampleBatchType, num_items: int) -> None:
             logger.info(msg)
 
 
-# TODO (artur): Remove ParallelIteratorWorker once we no longer support executionplans
 @DeveloperAPI
-class ReplayBuffer(ParallelIteratorWorker, FaultAwareApply):
+class ReplayBufferInterface(metaclass=ABCMeta):
+    """Abstract base class for all of RLlib's replay buffers.
+    
+    Mainly defines the `add()` and `sample()` methods that every buffer class
+    must implement to be usable by an Algorithm.
+    Buffers may determine on all the implementation details themselves, e.g.
+    whether to store single timesteps, episodes, or episode fragments or whether
+    to return fixed batch sizes or per-call defined ones.
+    """
+
+    @abstractmethod
+    @DeveloperAPI
+    def __len__(self) -> int:
+        """Returns the number of items currently stored in this buffer."""
+
+    @abstractmethod
+    @DeveloperAPI
+    def add(self, batch: SampleBatchType, **kwargs) -> None:
+        """Adds a batch of experiences or other data to this buffer.
+
+        Splits batch into chunks of timesteps, sequences or episodes, depending on
+        `self._storage_unit`. Calls `self._add_single_batch` to add resulting slices
```

Comment:
"Calls `self._add_single_batch` to add resulting slices
        to the buffer storage." may or may not be true for different implementations.

Reply:
Ah, yes, I'll move this back to `ReplayBuffer` and fix the more generic comment here to NOT include a mention of `self._add_single_batch`.

Reply:
done